lib/IRGen/IRABIDetailsProvider.cpp

```diff
@@ -122,6 +126,32 @@ class IRABIDetailsProviderImpl {
     return {returnTy, {paramTy}};
   }
 
+  llvm::SmallVector<IRABIDetailsProvider::ABIAdditionalParam, 1>
+  getFunctionABIAdditionalParams(AbstractFunctionDecl *afd) {
+    llvm::SmallVector<IRABIDetailsProvider::ABIAdditionalParam, 1> params;
+    Mangle::ASTMangler mangler;
+
+    auto function = SILFunction::getFunction(SILDeclRef(afd), *silMod);
+    auto mangledName = mangler.mangleAnyDecl(afd, /*prefix=*/true);
+
+    if (function->getName() == mangledName) {
```

Comment:
You don't need to compare the names here. You already know that the SIL function you have is this AST function :)